ci_framework/playbooks/07-deploy-edpm-with-bmaas.yml

```diff
@@ -0,0 +1,46 @@
+- hosts: "{{ cifmw_target_host | default('localhost') }}"
+  gather_facts: false
+  tasks:
+    - name: Call make_edpm_compute_baremetal target
+      ansible.builtin.include_role:
+        name: 'install_yamls_makes'
+        tasks_from: 'make_edpm_compute_baremetal'
+      vars:
+        make_edpm_compute_baremetal_env:
+          NETWORK_ISOLATION: false
+          OPENSTACK_IMG: "{{ cifmw_operator_build_output['operators']['openstack-operator']['image_catalog'] }}"
+
+    - name: Generate Bash Script to check if compute nodes are provisioned
+      ansible.builtin.copy:
+        dest: "{{ cifmw_basedir|default(ansible_user_dir ~ '/ci-framework') }}/artifacts/wait_for_bmaas_provisioning.sh"
+        content: |
+          #!/bin/bash
+          set -euo pipefail
+          set -x
+          timeout 600s bash -c 'until [ "$(oc get po -l  osp-provisionserver/name=openstackprovisionserver -n openstack -o name)" != "" ]; do sleep 1; done'
+          oc wait deployment openstackprovisionserver-openstackprovisionserver -n openstack --for condition=Available --timeout=600s
+          # Get the list of host names
+          host_names=$(oc get baremetalhost -n openstack -o=jsonpath='{.items[*].metadata.name}')
+          # Loop through each host name
+          for host_name in $host_names; do
+              echo "Waiting for host $host_name to reach the 'provisioned' state..."
+              oc wait bmh $host_name -n openstack --for=jsonpath='{.status.provisioning.state}'=provisioned --timeout=1200s
+              echo "Host $host_name has reached the 'provisioned' state."
+          done
+          echo "oc get bmh -n openstack"
+          oc get bmh -n openstack
+        mode: '0755'
+
+    - name: Run Bash Script to wait till compute nodes are provisioned
+      ansible.builtin.shell: "{{ cifmw_basedir|default(ansible_user_dir ~ '/ci-framework') }}/artifacts/wait_for_bmaas_provisioning.sh &> {{ cifmw_basedir|default(ansible_user_dir ~ '/ci-framework') }}/logs/wait_for_bmaas_provisioning.log"
+
+    - name: Patch OpenStackDataPlane
+      command: oc patch openstackdataplane openstack-edpm --type=merge -n openstack -p '{"spec":{"deployStrategy":{"deploy":true}}}'
+      
+    - name: Wait for OpenStackDataPlane to be Ready
+      command: oc wait --timeout=1800s --for=condition=Ready openstackdataplane openstack-edpm -n openstack
+
+    - name: Print OpenStackDataPlane status
+      ansible.builtin.shell: |
+         oc get openstackdataplane/openstack-edpm -o yaml -n openstack &>> "{{ cifmw_basedir|default(ansible_user_dir ~ '/ci-framework') }}/logs/openstackdataplane_status.log"
+         oc get openstackdataplanenode -n openstack &>> "{{ cifmw_basedir|default(ansible_user_dir ~ '/ci-framework') }}/logs/openstackdataplane_status.log"
```

Comment:
IMHO that should be in a role.

Reply:
Hi @cjeanner and @pablintino,

Thanks for your suggestion about moving this to a role. 

Basically, here we are waiting wait for 3 resources after running the `make_edpm_compute_baremetal` target.

* provisioner
* bmh 
* dataplane(after patching resource to start deploy) 
* log collection(can be moved in collect logs step)

About reusability, We can possibly divide, edpm_deploy role into two and "Wait for EDPM compute deployment to finish" part can be a separate role which can we called from "edpm_deploy"/ Or reuse the existing "edpm_deploy" if we have the right skips.

Currently, in the "edpm_deploy" [2], we only wait for "nova-edpm-compute-0-deploy-nova." I believe using "oc wait --timeout=1800s --for=condition=Ready openstackdataplane openstack-edpm -n openstack" would be a better check(which I have used here)

Perhaps we can address all of this in a follow-up?

Reply:
If the follow up is somehting that is gonna happen for me it's OK, if the follow up is never gonna happen or get delayed cause there are other priorities... maybe is better to do it in this PR. Depends if you expect to do it with some guarantees.

Reply:
Why can't we get that in this current PR? Usually "later change" never happens because of time constraints..

Reply:
Yep, getting a role would be a way to get a cleaner playbook - and, mostly, avoid creating "yet another playbook" as said in my previous comment. As for @pablintino comment, I'll let you and him sort this out. Getting the tasks in a role would allow a better modification later imho.
Note: in the role, you may then leverage the `dry_run` parameter from the ci_make (you can access it via the `vars` when you include_role "install_yamls_makes" - there are some existing examples already).

That way, molecule should be easy to get working.

Reply:
created a new role as suggested, marking this as resolved.